### DISCUSSION TITLE
WhatsApp automatically adding a number as suffix to device name in the list of Linked devices

### DISCUSSION THREAD

**Discussion by uandme72** [tulir/whatsmeow#667] [WhatsApp protocol Q&A]
WhatsApp for iOS adding a number as suffix to device name in the list of Linked devices. WhatsApp is automatically suffixing a number to the device name in the list of Linked devices shown in WhatsApp app settings. The suffixed number has no correlation to the actual number of linked devices at that time. For example, a number “2” is suffixed to the device name in list of Linked devices shown in WhatsApp, even though that computer/ device is the only one shown as a linked device. If we try to edit the device name to remove the suffixed digit, it does not move forward and save. 
Whatsapp support refuses to discuss or resolve the matter.

Please help.

**Comment by ghost** [comment]
You mean `<number>:<device_id>`  ?

**Comment by uandme72** [comment]
I am attaching a screenshot of WhatsApp for iOS, where it is showing a suffix added to the device name.
The surprising part is, even if there are no prior devices linked, the suffix does not vanish.
![IMG_0020](https://github.com/user-attachments/assets/d8490c61-6748-43cd-94ca-b6baa4e6909c)

**Comment by ghost** [comment]
Default OS name is `whatsmeow` when it unchanged.

https://github.com/tulir/whatsmeow/blob/main/store/clientpayload.go#L126
